### DISCUSSION TITLE
[vite] warning: unplugin-vue-better-define Error: EISDIR: illegal operation on a directory, read

### DISCUSSION THREAD

**Discussion by nigiwen** [vue-macros/vue-macros#773] [Q&A]
### 为什么一使用这个插件就报错，不使用是正常的
![image](https://github.com/user-attachments/assets/409f8291-5e78-4de4-96b3-9f214c07b4a6)
![image](https://github.com/user-attachments/assets/03b1b946-bffc-433b-919e-d9b8d80e656c)

**Comment by sxzz** [comment]
Let's track #774

**Comment by sxzz** [comment]
It should be fixed in `unplugin-vue-macros@2.11.6`. If the problem persists, please provide a minimal reproduction.

**Reply by nigiwen** [reply]
no fix

**Reply by sxzz** [reply]
Then provide a minimal reproduction repo

**Reply by nigiwen** [reply]
> Then provide a minimal reproduction repo

我尝试复现了可惜没复现出来，但是我发现类型导入时省略了`/index`就会报错，补全index就没报错了。

还有我想请问 下面这个问题怎么解决

![image](https://github.com/user-attachments/assets/ed26ac0a-1d99-4e9e-90b9-014c0f79d05d)

```ts
import type { TableProps as ElTableProps, TableColumnCtx } from 'element-plus'

export type TableProps<
  T extends object = object,
  Y extends object = object,
> =
  & Partial<ElTableProps<T>>
  & TableMoreProps<T, Y>

export interface TableMoreProps<
  Row extends object = object,
  Params extends object = object,
> {
  items: TableItems<Row>[]
  loading?: boolean
  pagination?: UsePaginationOptions
  api?:
    | ((params?: Params) => Promise<ResultData<ResPage<Row>>>)
    | ((params?: Params) => Promise<ResultData<Row[]>>)
    | Row[]
  apiParams?: Params
}
```
